### ISSUE TITLE
09_unsupervised_learning.ipynb   DBSCAN

### ISSUE THREAD

**Issue by satya400** [ageron/handson-ml2#79]
Hi ageron,

Thank you very much for this nice book.

I have a question in the 9th Chapter's code for DBSCAN - in the jupyter note book for this chapter, in cell [117] ,  do we need the 2nd scatter?  Doesn't it try to emphasize the core instances again?  Also, for the 2nd chart (with epsilon = 0.2), this 2nd scatter is creating an impression that there is a central dense zone and the outer zone is composed of  non-core instances? But for this case, the non-core instance count seems to be zero.

Please review and guide me if I am missing anything.

Thanks
Satya

**Comment by ageron** [comment]
Hi @satya400 ,

Sorry for the late response, I was busy moving to NZ, but the current lockdown is giving me time to unpile github issues at last!

I used two scatters for core instances because the first one draws circles of the appropriate size to represent the core instance's ε-neighborhood, while the second one places a small star * in the middle of the circle, just to make it clear that there's a core instance there. The non-core instances are just represented as dots. By definition, non-core instances always belong to the ε-neighborhood of at least one core instance. And the anomalies are represented as crosses. By definition, they never belong to the ε-neighborhood of a core instance.

To answer your second question, in the 2nd scatter plot, the ε-neighborhood radius is greater than in the 1st scatter plot, so instances have more instances in their ε-neighborbood, and thus many instances that were non-core (i.e., dots) in the 1st scatter plot have become core instances (i.e., stars) in the 2nd scatter plot. In fact, all of them have become core instances, so there's no non-core instance left, as you rightly point out.

In short, the colored region does not represent non-core instances, it represents the ε-neighborhoods of core instances.

Hope this helps (please close the issue if it's all clear now).

**Comment by satya400** [comment]
Hi @ageron 

  Thanks for the detailed reply.  Hope you are comfortably settled now in NZ.
